src/client/app/actions/lineReadings.js

```diff
@@ -5,43 +5,61 @@
  */
 
 import axios from 'axios';
+import { DATA_TYPE_METER, DATA_TYPE_GROUP } from '../utils/Datasources';
 
 export const REQUEST_LINE_READINGS = 'REQUEST_LINE_READINGS';
 export const RECEIVE_LINE_READINGS = 'RECEIVE_LINE_READINGS';
 
 /**
  * @param {State} state
- * @param {number} meterID
+ * @param {number} dsID
  * @param {TimeInterval} timeInterval
+ * @param {String} type either DATA_TYPE_METER, DATA_TYPE_GROUP
  */
-function shouldFetchLineReadings(state, meterID, timeInterval) {
-	const readingsForMeterID = state.readings.line.byMeterID[meterID];
-	if (readingsForMeterID === undefined) {
+function shouldFetchLineReadings(state, dsID, timeInterval, type) {
+	let dsArray;
+	if (type === DATA_TYPE_GROUP) {
+		dsArray = state.readings.line.byGroupID;
+	} else if (type === DATA_TYPE_METER) {
+		dsArray = state.readings.line.byMeterID;
+	}
+
+	const readingsForID = dsArray[dsID];
+	if (readingsForID === undefined) {
 		return true;
 	}
-	if (readingsForMeterID[timeInterval] === undefined) {
+	if (readingsForID[timeInterval] === undefined) {
 		return true;
 	}
-	const readingsForTimeInterval = state.readings.line.byMeterID[meterID][timeInterval];
+	const readingsForTimeInterval = dsArray[dsID][timeInterval];
 	return readingsForTimeInterval === undefined && !readingsForTimeInterval.isFetching;
 }
 
-function requestLineReadings(meterIDs, timeInterval) {
-	return { type: REQUEST_LINE_READINGS, meterIDs, timeInterval };
+function requestLineReadings(dsIDs, timeInterval, dstype) {
+	return { type: REQUEST_LINE_READINGS, dsIDs, timeInterval, dstype };
 }
 
-function receiveLineReadings(meterIDs, timeInterval, readings) {
-	return { type: RECEIVE_LINE_READINGS, meterIDs, timeInterval, readings };
+function receiveLineReadings(dsIDs, timeInterval, readings, dstype) {
+	return { type: RECEIVE_LINE_READINGS, dsIDs, timeInterval, readings, dstype };
 }
 
-function fetchLineReadings(meterIDs, timeInterval) {
+function fetchLineReadings(dsIDs, timeInterval, dstype) {
 	return dispatch => {
-		dispatch(requestLineReadings(meterIDs, timeInterval));
+		dispatch(requestLineReadings(dsIDs, timeInterval, dstype));
 		// The api expects the meter ids to be a comma-separated list.
-		const stringifiedMeterIDs = meterIDs.join(',');
-		return axios.get(`/api/readings/line/${stringifiedMeterIDs}`, {
+		const stringifiedIDs = dsIDs.join(',');
+		let endpoint;
+		if (dstype === DATA_TYPE_GROUP) {
+			endpoint = '/api/readings/line/groups';
+		} else if (dstype === DATA_TYPE_METER) {
+			endpoint = '/api/readings/line/meters';
+		} else {
+			console.error('Unknown datatype requested in fetchLineReadings: ', dstype);
+			endpoint = '/api/nonexistant';
```

Comment:
Perhaps just return Promise.resolve() here instead of sending an unnecessary request to a non-existant endpoint.